src/index.ts

```diff
@@ -62,3 +62,4 @@ export * as AutoLogging from "@hyperion/hyperion-autologging/src/AutoLogging";
 export { ALSurfaceCapability } from "@hyperion/hyperion-autologging/src/ALSurface";
 export * as ALSurfaceUtils from "@hyperion/hyperion-autologging/src/ALSurfaceUtils";
 export * as ALCustomEvent from '@hyperion/hyperion-autologging/src/ALCustomEvent';
+export { getReactComponentData_THIS_CAN_BREAK } from '@hyperion/hyperion-autologging/src/ALReactUtils';
```

Comment:
I like the idea of passing to component validator,  but I'm leaning towards adding another option in the UIEventConfig,  because I don't necessarily want this to apply to all events we collect.  In this case we know hover is noisy and want to test a hypothesis that it is the large stack walking we are doing but I don't want to blanket cap the lookup for every event that uses this method.

Reply:
That's actually a good approach. So do you mean instead of `cacheElementReactInfo` being boolean, you can have an integer option for `elementReactInfoDepth` that -1 means all, 0 means none and positive integers determine the stack limit?